### ISSUE TITLE
Compaction not work with attrs option

### ISSUE THREAD

**Issue by fisuda** [Fiware/context.Orion-LD#735]
The behavior of `GET /ngsi-ld/v1/entities/{id}?attrs={attr}` seems to be changed bettween `0.6.1` and `post-v0.6`.
The compaction does not work with attrs option.

The following is a test script.

```
#!/bin/sh
curl -s -L -X GET 'http://orion-ld:1026/version'

curl -s -iX POST 'http://orion-ld:1026/ngsi-ld/v1/entities/' \
-H 'Content-Type: application/json' \
-H 'Link: <https://context.lab.letsfiware.jp/ngsi-context.jsonld>; rel="http://www.w3.org/ns/json-ld#context"; type="application/ld+json"' \
--data-raw '{
      "id": "urn:ngsi-ld:TemperatureSensor:001",
      "type": "TemperatureSensor",
      "category": {
            "type": "Property",
            "value": "sensor"
      },
      "temperature": {
            "type": "Property",
            "value": 25,
            "unitCode": "CEL"
      }
}'

curl -s -X GET 'http://orion-ld:1026/ngsi-ld/v1/entities/urn:ngsi-ld:TemperatureSensor:001' \
-H 'Link: <https://context.lab.letsfiware.jp/ngsi-context.jsonld>; rel="http://www.w3.org/ns/json-ld#context"; type="application/ld+json"' | jq .

curl -s -G -X GET 'http://orion-ld:1026/ngsi-ld/v1/entities/urn:ngsi-ld:TemperatureSensor:001' \
-H 'Link: <https://context.lab.letsfiware.jp/ngsi-context.jsonld>; rel="http://www.w3.org/ns/json-ld#context"; type="application/ld+json"' \
-d 'attrs=temperature' | jq .
```

The result of running it in `post-v0.6` is shown:
```
{
  "orionld version": "post-v0.6",
  "orion version":   "1.15.0-next",
  "uptime":          "0 d, 0 h, 0 m, 19 s",
  "git_hash":        "nogitversion",
  "compile_time":    "Sat Feb 27 12:36:35 UTC 2021",
  "compiled_by":     "root",
  "compiled_in":     "buildkitsandbox",
  "release_date":    "Sat Feb 27 12:36:35 UTC 2021",
  "doc":             "https://fiware-orion.readthedocs.org/en/master/"
}

HTTP/1.1 201 Created
Connection: Keep-Alive
Content-Length: 0
Location: /ngsi-ld/v1/entities/urn:ngsi-ld:TemperatureSensor:001
Date: Sat, 27 Feb 2021 23:34:46 GMT

{
  "@context": "https://context.lab.letsfiware.jp/ngsi-context.jsonld",
  "id": "urn:ngsi-ld:TemperatureSensor:001",
  "type": "TemperatureSensor",
  "category": {
    "type": "Property",
    "value": "sensor"
  },
  "temperature": {
    "type": "Property",
    "value": 25,
    "unitCode": "CEL"
  }
}

{
  "@context": "https://context.lab.letsfiware.jp/ngsi-context.jsonld",
  "id": "urn:ngsi-ld:TemperatureSensor:001",
  "type": "TemperatureSensor",
  "https://w3id.org/saref#temperature": {
    "type": "Property",
    "value": 25,
    "unitCode": "CEL"
  }
}
```

The result of running it in `0.6.1` is shown:
```
{
  "orionld version": "0.6.1",
  "orion version":   "1.15.0-next",
  "uptime":          "0 d, 0 h, 0 m, 22 s",
  "git_hash":        "nogitversion",
  "compile_time":    "Thu Jan 28 10:52:23 UTC 2021",
  "compiled_by":     "root",
  "compiled_in":     "buildkitsandbox",
  "release_date":    "Thu Jan 28 10:52:23 UTC 2021",
  "doc":             "https://fiware-orion.readthedocs.org/en/master/"
}

HTTP/1.1 201 Created
Connection: Keep-Alive
Content-Length: 0
Location: /ngsi-ld/v1/entities/urn:ngsi-ld:TemperatureSensor:001
Date: Sat, 27 Feb 2021 23:37:12 GMT

{
  "@context": "https://context.lab.letsfiware.jp/ngsi-context.jsonld",
  "id": "urn:ngsi-ld:TemperatureSensor:001",
  "type": "TemperatureSensor",
  "category": {
    "type": "Property",
    "value": "sensor"
  },
  "temperature": {
    "type": "Property",
    "value": 25,
    "unitCode": "CEL"
  }
}

{
  "@context": "https://context.lab.letsfiware.jp/ngsi-context.jsonld",
  "id": "urn:ngsi-ld:TemperatureSensor:001",
  "type": "TemperatureSensor",
  "temperature": {
    "type": "Property",
    "value": 25,
    "unitCode": "CEL"
  }
}
```

Docker images
```
$ docker images | grep orion-ld
fiware/orion-ld                          0.7-PRE-71                     d82d77013395   11 hours ago     978MB
fiware/orion-ld                          0.6.1                          90d6b3e8f194   4 weeks ago      978MB
```

**Comment by kzangeli** [comment]
ok!

Saw it and yes, this is a bug - I actually fixed it already - PR is on it's way.

I recently had to modify the behaviour of 'attrs' for geo-json mime type and I managed (as so often) to destroy something else in the process.
In this case I had forgotten to copy a string before modifying it, and thus modified (destroyed) an item in the @context.

Thank you again for finding bugs in the broker - it's a more than welcome help!

**Comment by kzangeli** [comment]
Should be fixed in PR #736

**Comment by fisuda** [comment]
I confirmed that this issue has been fixed. Thank you for your prompt response.

```
$ docker images | grep orion-ld
fiware/orion-ld                          0.7-PRE-73                     bbb310b77759   11 minutes ago   978MB
```

```
{
  "orionld version": "post-v0.6",
  "orion version":   "1.15.0-next",
  "uptime":          "0 d, 0 h, 5 m, 43 s",
  "git_hash":        "nogitversion",
  "compile_time":    "Sun Feb 28 10:19:18 UTC 2021",
  "compiled_by":     "root",
  "compiled_in":     "buildkitsandbox",
  "release_date":    "Sun Feb 28 10:19:18 UTC 2021",
  "doc":             "https://fiware-orion.readthedocs.org/en/master/"
}

HTTP/1.1 201 Created
Connection: Keep-Alive
Content-Length: 0
Location: /ngsi-ld/v1/entities/urn:ngsi-ld:TemperatureSensor:001
Date: Sun, 28 Feb 2021 10:38:59 GMT

{
  "@context": "https://context.lab.letsfiware.jp/ngsi-context.jsonld",
  "id": "urn:ngsi-ld:TemperatureSensor:001",
  "type": "TemperatureSensor",
  "category": {
    "type": "Property",
    "value": "sensor"
  },
  "temperature": {
    "type": "Property",
    "value": 25,
    "unitCode": "CEL"
  }
}

{
  "@context": "https://context.lab.letsfiware.jp/ngsi-context.jsonld",
  "id": "urn:ngsi-ld:TemperatureSensor:001",
  "type": "TemperatureSensor",
  "temperature": {
    "type": "Property",
    "value": 25,
    "unitCode": "CEL"
  }
}
```
